lint/inverseMethodCall_checker.go

```diff
@@ -0,0 +1,57 @@
+package lint
+
+import (
+	"go/ast"
+	"go/types"
+	"strings"
+)
+
+func init() {
+	addChecker(&inverseMethodCallChecker{}, attrExperimental)
+}
+
+type inverseMethodCallChecker struct {
+	checkerBase
+}
+
+func (c *inverseMethodCallChecker) InitDocumentation(d *Documentation) {
+	d.Summary = "Detects inverse method execution"
+	d.Before = `type foo struct{}
+f := foo{}
+foo.bar(f)`
+	d.After = `type foo struct{}
+f := foo{}
+f.bar()`
+}
+
+func (c *inverseMethodCallChecker) VisitExpr(x ast.Expr) {
+	call, ok := x.(*ast.CallExpr)
+	if !ok {
+		return
+	}
+
+	s, ok := call.Fun.(*ast.SelectorExpr)
+	if !ok {
+		return
+	}
+
+	id, ok := s.X.(*ast.Ident)
+	if !ok {
+		return
+	}
```

Comment:
^^^ all this can be exported to a method

Reply:
We have `identOf`. It can be extended, if you want it.
But it should probably be done in a separate PR.